bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -258,17 +258,29 @@ final class BagsManager(executorService: ExecutorService) {
   private var listeners: List[BagsStatusListener] = Nil
 
   def addBagsStatusListener(l: BagsStatusListener): Unit =
-    if (!listeners.contains(l))
-      listeners = l :: listeners
+    listeners.synchronized {
```

Comment:
We shouldn't synchronize on the object that `listeners` references since it is immutable and the reference is going to change anyway.  Probably the lock associated with `BagsManager` itself is what you want.

````scala
synchronized {
....
}
````